packages/example/lib/useConnectionStatus.ts

```diff
@@ -0,0 +1,27 @@
+import { useSession } from 'next-auth/react';
```

Comment:
I am worried about this hook combinoing both connection & auth state as consumers may use it as a way to retrieve the source of truth for combined connection & authentication state. These states can get very complex & I am also not sure how I feel about `!isConnected === 'disconnected'` as `!isConnected` could mean either `connecting`, `reconnecting` or `disconnected` from wagmi. 

Wonder if we can just use `useAccount` & `useSession` hook directly in the usage? If consumers want to abstract this, they can make their own `useConnectionStatus` hook like RainbowKit has done in `src/hooks`. 

```tsx
const { status } = useSession()

const ready = mounted && (authEnabled ? status !== 'loading' : true);
const connected = authEnabled ? status === 'authenticated' : isConnected;
```

Thoughts?

Reply:
I agree, I think it makes sense to avoid giving users too much abstraction in the example. I'll take a look.